library/Vanilla/Models/SiteMeta.php

```diff
@@ -149,10 +155,27 @@ public function __construct(
         // DeploymentCacheBuster
         $this->cacheBuster = $deploymentCacheBuster->value();
 
+        $this->session = $session;
+
         // Theming
-        $themeKey = $themeModel->getViewThemeKey();
-        $this->activeThemeKey = $themeKey;
-        $this->activeThemeViewPath =  $themeModel->getThemeViewPath($themeKey);
+        $themeID = null;
+        $sectionThemeID = '';
+        $configParam = 'Garden.Theme';
+        if ($this->currentSiteSection->getSectionThemeID()) {
+            $sectionThemeID =  $this->currentSiteSection->getSectionThemeID();
+            $configParam = 'Garden.CurrentTheme';
+        }
+
+        $theme = $themeModel->getCurrentTheme($sectionThemeID);
+        $themeID = $theme['themeID'] ?? null;
+        $themeAddon = $themeModel->getCurrentThemeAddon($configParam);
```

Comment:
If we're just getting the current whats the scenario where we want current, but actually some other themeID instead. We should be able to just use `getCurrentThemeAddon()`.

Most of this logic should probably be over in the theme model, instead of here.